lms/djangoapps/courseware/tests/test_inline_analytics_integration.py

```diff
@@ -35,32 +35,35 @@ def setUp(self):
         json_analytics_data = json.dumps(analytics_data)
         self.data = {"data": json_analytics_data}
 
-    @override_settings(ANALYTICS_ANSWER_DIST_URL='dummy_url')
+    @override_settings(ANALYTICS_DATA_URL='dummy_url')
+    @override_settings(ZENDESK_URL='https://stanfordonline.zendesk.com')
     def test_regular_user(self):
 
         request = self.factory.post('', self.data)
         request.user = self.user
 
         response = get_analytics_answer_dist(request)
-        self.assertEquals(response.content, 'A problem has occurred retrieving the data, please report the problem.')
+        self.assertEquals(response.content, 'A problem has occurred retrieving the data, to report the problem click <a href="https://stanfordonline.zendesk.com/hc/en-us/requests/new">here</a>')
 
+    @override_settings(ZENDESK_URL='https://stanfordonline.zendesk.com')
     def test_no_url(self):
 
         request = self.factory.post('', self.data)
         request.user = self.instructor
 
         response = get_analytics_answer_dist(request)
-        self.assertEquals(response.content, 'A problem has occurred retrieving the data, please report the problem.')
+        self.assertEquals(response.content, 'A problem has occurred retrieving the data, to report the problem click <a href="https://stanfordonline.zendesk.com/hc/en-us/requests/new">here</a>')
 
-    @override_settings(ANALYTICS_ANSWER_DIST_URL='dummy_url')
-    @patch('urllib2.urlopen')
+    @override_settings(ANALYTICS_DATA_URL='dummy_url')
+    @override_settings(ZENDESK_URL='https://stanfordonline.zendesk.com')
     @patch('courseware.views.process_analytics_answer_dist')
-    def test_staff_and_url(self, mock_process_analytics, mock_requests):
+    @patch('courseware.views.Client')
+    def test_staff_and_url(self, mock_client, mock_process_analytics):
 
-        mock_resp = MagicMock()
-        mock_read = MagicMock(return_value="{}")
-        mock_resp.read = mock_read
-        mock_requests.return_value = mock_resp
+        mock_client.return_value.modules.return_value.answer_distribution.return_value = [
+            {
+            },
+        ]
```

Comment:
Seems simpler to just have this on one line, since the list is empty.
`mock_client.return_value.modules.return_value.answer_distribution.return_value = [{}]`